src/@apollo/client/react/hooks/ApolloClient__React_UseMutation.re

```diff
@@ -0,0 +1,368 @@
+module ApolloClient = ApolloClient__ApolloClient;
+module ApolloError = ApolloClient__ApolloError;
+module ErrorPolicy = ApolloClient__Core_WatchQueryOptions.ErrorPolicy;
+module FetchPolicy__noCacheExtracted = ApolloClient__Core_WatchQueryOptions.FetchPolicy__noCacheExtracted;
+module Graphql = ApolloClient__Graphql;
+module MutationHookOptions = ApolloClient__React_Types.MutationHookOptions;
+module MutationTuple = ApolloClient__React_Types.MutationTuple;
+module MutationTuple__noVariables = ApolloClient__React_Types.MutationTuple__noVariables;
+module MutationTuple__optionalVariables = ApolloClient__React_Types.MutationTuple__optionalVariables;
+module MutationUpdaterFn = ApolloClient__Core_WatchQueryOptions.MutationUpdaterFn;
+module OperationVariables = ApolloClient__Core_Types.OperationVariables;
+module RefetchQueryDescription = ApolloClient__Core_WatchQueryOptions.RefetchQueryDescription;
+module Types = ApolloClient__Types;
+module Utils = ApolloClient__Utils;
+
+module type Operation = Types.Operation;
+module type OperationNoRequiredVars = Types.OperationNoRequiredVars;
+
+module Js_ = {
+  // export declare function useMutation<TData = any, TVariables = OperationVariables>(mutation: DocumentNode, options?: MutationHookOptions<TData, TVariables>): MutationTuple<TData, TVariables>;
+  [@bs.module "@apollo/client"]
+  external useMutation:
+    (
+      . ~mutation: Graphql.documentNode,
+      ~options: MutationHookOptions.Js_.t('jsData, 'variables)=?
+    ) =>
+    MutationTuple.Js_.t('jsData, 'variables) =
+    "useMutation";
+};
+
+let useMutation:
+  type data jsData jsVariables.
+    (
+      ~awaitRefetchQueries: bool=?,
+      ~context: Js.Json.t=?,
+      ~client: ApolloClient.t=?,
+      ~errorPolicy: ErrorPolicy.t=?,
+      ~fetchPolicy: FetchPolicy__noCacheExtracted.t=?,
+      ~ignoreResults: bool=?,
+      ~notifyOnNetworkStatusChange: bool=?,
+      ~onError: ApolloError.t => unit=?,
+      ~onCompleted: data => unit=?,
+      ~optimisticResponse: jsVariables => data=?,
+      ~refetchQueries: RefetchQueryDescription.t=?,
+      ~update: MutationUpdaterFn.t(data)=?,
+      (module Operation with
+         type t = data and
+         type Raw.t = jsData and
+         type Raw.t_variables = jsVariables)
+    ) =>
+    MutationTuple.t(data, jsVariables) =
+  (
+    ~awaitRefetchQueries=?,
+    ~context=?,
+    ~client=?,
+    ~errorPolicy=?,
+    ~fetchPolicy=?,
+    ~ignoreResults=?,
+    ~notifyOnNetworkStatusChange=?,
+    ~onError=?,
+    ~onCompleted=?,
+    ~optimisticResponse=?,
+    ~refetchQueries=?,
+    ~update=?,
+    (module Operation),
+  ) => {
+    let jsMutationTuple =
+      Js_.useMutation(.
+        ~mutation=Operation.query->Utils.castStringAsDocumentNode,
```

Comment:
@jfrolich Any plans to add a `template-tag-return-type`?

Reply:
Oh you are using template tag now? Hmm I was thinking making the query type opaque in case of using a template tag. So then it's up to the libraries to cast.

Reply:
Hmm yes specifying a return type in configuration would be better indeed!

Reply:
@jfrolich Awesome, looking forward to it.